### ISSUE TITLE
Restore icons after sleep

### ISSUE THREAD

**Issue by Rigido71** [com-entonos/Desktop-Icon-Manager#1]
Hi,
starting with Sonoma (I'm pretty sure) my external disk icons started moving from the rightmost side of the only attached screen somewhere else on the desktop. I didn't understand WHEN this happen but it would be great if DIM could restore icons on wake-up.
![BEFORE](https://github.com/com-entonos/Desktop-Icon-Manager/assets/29772377/64a4f2fc-db73-404a-a99e-0f181eb39ede)
![AFTER](https://github.com/com-entonos/Desktop-Icon-Manager/assets/29772377/7a858f75-fc28-4b39-baf3-2007892df88a)

**Comment by entonos** [comment]
First I've heard of this. 

One could add an observer for screen wake notification. Something like:
`NSWorkspace.shared.notificationCenter.addObserver(forName: NSWorkspace.screensDidWakeNotification, object: nil, queue: .main, using: {_ in self.restore(currentName) })`

**Comment by entonos** [comment]
the new update (4.3.1) does an automatic Restore Icon Positions 10 seconds after wake if DIM is running.

**Comment by MeAndMyRhythmbox** [comment]
Could something similar be added to auto-restore whenever screen RESOLUTION changes? 

I use a KVM switch to toggle my main display between Mac and work laptop. However, because I have a small 2nd display the Mac moves all icons onto it, thinking the main display has been unplugged. When I switch back from work laptop to Mac, the Mac sees 2 screens again, but makes a mess of any icons on the right hand side! This only happened after upgrading from macOS Monterey to Sequoia and is driving me nuts!

**Comment by entonos** [comment]
Yes. There is a notification for screen changes which DIM could use. For it to work, DIM would need to be running at the time you activate the KVM. 

In the meanwhile, just start DIM when you need to Restore the icon positions and hit Restore. Regardless of how many screens there are, the Finder will define a 'rectangle' large enough to cover all screens and Dim will scale the icons' positions to fit into that rectangle. This may not be what you want (the Finder will not place an icon in a position that's not on either screen, but will place it somewhere depending on the version of macOS and what is currently on the Desktop- it is the Finder that moves the icons, DIM only keeps track of the icons and asks the Finder to place them- see DIM.applescript). In that case, create a second icon Arrangement (i.e. gear menu>Edit...>+ to add a new Arrangement). One for one screen and another for two screens. Create these two Arrangements when you have first placed the icons where you would like them, respectively.

This particular problem has come up before. One could, I suppose, look at all the saved Arrangements and then choose which one makes sense to use (based on the number, sizes and relative positions of the displays). But that has problems since it's not guaranteed to be unique.

**Comment by MeAndMyRhythmbox** [comment]
Thanks for detailed response. I'm not too worried about where the icons go on 2nd screen (and can of course use a second stored arrangement). I just wish icons would come back on the main screen when I switch back, in the same place they were originally. That was the behavior in Monterrey (and Mojave previously) before I upgraded to Sequoia - thankfully your app exists to help work around this :)

**Comment by entonos** [comment]
OK. I can set up a notification for DIM to do this.

**Comment by MeAndMyRhythmbox** [comment]
Excellent, thank you so much 👍
